assets/js/pagination.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const CURRENT_CONTINUATION = (new URL(document.location)).searchParams.get("continuation");
+const CONT_CACHE_KEY = `continuation_cache_${encodeURIComponent(window.location.pathname)}`;
+
+function get_data(){
+    return JSON.parse(sessionStorage.getItem(CONT_CACHE_KEY)) || [];
+}
+
+function save_data(){
+    const prev_data = get_data();
+    prev_data.push(CURRENT_CONTINUATION);
+
+    sessionStorage.setItem(CONT_CACHE_KEY, JSON.stringify(prev_data));
+}
+
+function button_press(){
+    let prev_data = get_data();
+    if (!prev_data.length) return null;
+
+    // Sanity check. Nowhere should the current continuation token exist in the cache 
+    // but it can happen when using the browser's back feature. As such we'd need to travel
+    // back to the point where the current continuation token first appears in order to 
+    // account for the rewind.
+    const conflict_at = prev_data.indexOf(CURRENT_CONTINUATION);
+    if (conflict_at != -1) {
+        prev_data.length = conflict_at;
+    }
+
+    const prev_ctoken = prev_data.pop();
+
+    // On the first page, the stored continuation token is null.
+    if (prev_ctoken === null) {
+        sessionStorage.removeItem(CONT_CACHE_KEY);
+        window.location.href = window.location.href.split('?')[0];
```

Comment:
If I understand that correctly, you're nuking all query parameters from the URL.
I'm pretty sure there are ways to selectively remove the `continuation` param only.

Reply:
The `removeItem` function only removes `CONT_CACHE_KEY` which is the current continuation array. It is defined as `continuation_cache_${encodeURIComponent(window.location.pathname)}`.

Reply:
I'm talking about the `window.location.href` part ^^

Reply:
There is a way to selectively remove just the `continuation` parameter, but it's not needed. It took me way too long to realize this, but when we go to the next page, we are already nuking all of the query parameters. Only the continuation parameter is saved. Let me know if you still want this changed.

Reply:
That behavior is actually a bug, tracked in #2130.
So that would be great if you could help make it less bad x)

Reply:
I fixed it (for the channel page only) in [e8c2388](https://github.com/iv-org/invidious/pull/4123/commits/e8c2388589494f1370231154ffb2030c3871291b).